chia-bls/src/gtelement.rs

```diff
@@ -0,0 +1,159 @@
+use blst::*;
+use chia_traits::chia_error::Result;
+use chia_traits::{read_bytes, Streamable};
+use sha2::{Digest, Sha256};
+use std::fmt;
+use std::hash::{Hash, Hasher};
+use std::io::Cursor;
+use std::mem::MaybeUninit;
+use std::ops::MulAssign;
+
+#[cfg(feature = "py-bindings")]
+use chia_py_streamable_macro::PyStreamable;
+#[cfg(feature = "py-bindings")]
+use chia_traits::from_json_dict::FromJsonDict;
+#[cfg(feature = "py-bindings")]
+use chia_traits::to_json_dict::ToJsonDict;
+#[cfg(feature = "py-bindings")]
+use pyo3::exceptions::PyValueError;
+#[cfg(feature = "py-bindings")]
+use pyo3::{pyclass, pymethods, IntoPy, PyAny, PyObject, PyResult, Python};
```

Comment:
It'd be preferable if the Python bindings weren't interspersed with the implementation code. This is a problem we have in `chia-protocol` with `PyStreamable` as well, for example. If we were to use a newtype, we could separate the bindings out and add them to the `wheel` crate instead. Much easier to maintain and expand to other language bindings in the future, but we'd need thorough tests to prevent breakage.

Reply:
the main challenge with that approach is that it would spread to all other chia classes and messages. e.g. most of `chia-protocol`. I think we would need to generate the python types by a macro in order for it to be practical.

The other risk is that objects would need to be copied in- and out of the python versions, which may be expensive in certain cases. like `FullBlock` for instance

Reply:
They can't be moved?